libs/sof/sof_multicamera_gpu.cpp

```
@@ -66,60 +68,112 @@ void MultiSOFGPU::LaunchTrackingPrimaryToSecondary(CameraId primary_id, CameraId
+  const size_t n = primary_obs.size();
+  auto& uvL = pair.uvL;
+  auto& cands = pair.cands;
+  auto& winners = pair.winners;
+  uvL.resize(n);
+  cands.resize(n);  // inner vectors retain their allocation across frames
+  // `assign` rather than `resize`: guarantees every slot starts as a zero-init `TrackData`, so a
+  // newly grown vector element cannot leak stale fields into a later `winners[i] = ...` copy.
+  winners.assign(n, TrackData{});
+
+  size_t max_candidates = 0;
+  for (size_t i = 0; i < n; ++i) {
```

Clang (14.0.6)

[note] +1, including nesting penalty of 0, nesting level increased to 1

(clang)